docs/set.rst

```diff
@@ -0,0 +1,471 @@
+
+
+Common API Functions
+--------------------
+
+.. TIP::
+   All of these functions that work for ``Set`` will also work for ``IntSet``,
+   which has the element type ``a`` specialized to ``Int``. Anywhere that you
+   see ``Set Int`` you can replace it with ``IntSet``.
+
+.. NOTE::
+   ``fromList [some,list,elements]`` is how a ``Set`` is printed.
+
+
+Construction and Conversion
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create an empty set
+"""""""""""""""""""
+
+::
+
+    Set.empty :: Set a
+    Set.empty = ...
+
+:haddock_short:`Data.Set#empty` creates a set with zero elements.
+
+::
+
+    Set.empty
+    > fromList []
+
+Create a set with one element (singleton)
+"""""""""""""""""""""""""""""""""""""""""
+
+::
+
+    Set.singleton :: a -> Set a
+    Set.singleton x = ...
+
+:haddock_short:`Data.Set#singleton` creates a set with a single element ``x`` in
+it.
+
+::
+
+    Set.singleton "containers"
+    > fromList ["containers"]
+
+    Set.singleton 1
+    > fromList [1]
+
+Create a set from a list
+""""""""""""""""""""""""
+
+::
+
+    Set.fromList :: Ord a => [a] -> Set a
+    Set.fromList xs = ...
+
+:haddock_short:`Data.Set#fromList` creates a set containing the elements of the
+list ``xs``. Since sets don't contain duplicates, if there are repeated elements
+in the list they will only appear once.
+
+::
+
+    Set.fromList ["base", "containers", "QuickCheck"]
+    > fromList ["QuickCheck","base","containers"]
+
+    Set.fromList [1, 1, 2, 3, 4, 4, 5, 1]
+    > fromList [1,2,3,4,5]
+
+Create a list from a set
+""""""""""""""""""""""""
+
+::
+
+    Set.toAscList, Set.toList, Set.elems :: Set a -> [a]
+    Set.toAscList s = ...
+
+:haddock_short:`Data.Set#toAscList`, :haddock_short:`Data.Set#toList`, and
+:haddock_short:`Data.Set#elems` return a list containing the elements of the set
+:haddock_short:``s`` in *ascending* order.
+
+.. NOTE::
+   These all do the same thing; use ``toAscList`` because its name indicates the
+   ordering.
+
+::
+
+    Set.toDescList :: Set a -> [a]
+    Set.toDescList s = ...
+
+:haddock_short:`Data.Set#toDescList` returns a list containing the elements of
+the set ``s`` in *descending* order.
+
+::
+
+    Set.toAscList (Set.fromList [0, 2, 4, 6])
+    > [0,2,4,6]
+
+    Set.toDescList (Set.fromList [0, 2, 4, 6]
+    > [6,4,2,0]
+
+
+Querying
+^^^^^^^^
+
+Check if a set is empty
+"""""""""""""""""""""""
+
+::
+
+    Set.null :: Set a -> Bool
+    Set.null s = ...
+
+:haddock_short:`Data.Set#null` returns ``True`` if the set ``s`` is empty,
+``False`` otherwise.
+
+::
+
+    Set.null Set.empty
+    > True
+
+    Set.null (Set.fromList [0, 2, 4, 6])
+    > False
+
+
+The number of elements in a set
+"""""""""""""""""""""""""""""""
+
+::
+
+    Set.size :: Set a -> Int
+    Set.size s = ...
+
+:haddock_short:`Data.Set#size` returns the number of elements in the set ``s``.
+
+::
+
+    Set.size Set.empty
+    > 0
+
+    Set.size (Set.fromList [0, 2, 4, 6])
+    > 4
+
+Check if an element is in a set (member)
```

Comment:
This should float to the top of the querying section; it's the most important query by far.

Reply:
Done.